agent/http.go

```diff
@@ -359,8 +359,9 @@ func parseWait(resp http.ResponseWriter, req *http.Request, b *structs.QueryOpti
 
 // parseConsistency is used to parse the ?stale and ?consistent query params.
 // Returns true on error
-func parseConsistency(resp http.ResponseWriter, req *http.Request, b *structs.QueryOptions) bool {
+func (s *HTTPServer) parseConsistency(resp http.ResponseWriter, req *http.Request, b *structs.QueryOptions) bool {
 	query := req.URL.Query()
+	b.AllowStale = !b.RequireConsistent && s.agent.config.HttpApiConfig.AllowStale
```

Comment:
Maybe I'm missing something obvious but I don't think the other version is correct. @preetapan What do you think?